nomad/resource_node_pool.go

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package nomad
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-nomad/nomad/helper"
+	"github.com/hashicorp/terraform-provider-nomad/nomad/helper/pointer"
+)
+
+const (
+	nodePoolMemoryOversubscriptionEnabled  = "enabled"
+	nodePoolMemoryOversubscriptionDisabled = "disabled"
+)
+
+func resourceNodePool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNodePoolWrite,
+		Update: resourceNodePoolWrite,
+		Delete: resourceNodePoolDelete,
+		Read:   resourceNodePoolRead,
+		Exists: resourceNodePoolExists,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Unique name for this node pool.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"description": {
```

Comment:
Do we want to add validation via `validation.StringLenBetween` here to avoid API calls that will just fail?

Reply:
Added in https://github.com/hashicorp/terraform-provider-nomad/pull/340/commits/a5fcae001420ded22f00fc1431c77903a116775e